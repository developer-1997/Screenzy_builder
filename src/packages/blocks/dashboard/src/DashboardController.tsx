import { IBlock } from "../../../framework/src/IBlock";
import { Message } from "../../../framework/src/Message";
import { BlockComponent } from "../../../framework/src/BlockComponent";
import MessageEnum, {
  getName
} from "../../../framework/src/Messages/MessageEnum";
import { runEngine } from "../../../framework/src/RunEngine";

// Customizable Area Start
// @ts-ignore
import RNAdvertisingId from "react-native-advertising-id";
import Toast from "react-native-simple-toast";
import { createRef } from "react";
// Customizable Area End

export const configJSON = require("./config.js");

export interface Props {
  navigation: any;
  id: string;
  // Customizable Area Start
  // Customizable Area End
}
interface S {
  // Customizable Area Start
  dashboardData: any;
  token: string;
  errorMsg: string;
  loading: boolean;
  count: number;
  quizeData: any;
  sound: any;
  quizQuestionLoaded: boolean;
  quizQuestionData: any;
  events: any;
  offline: boolean;
  offlineVideo: any;
  quizid: any;
  userName: string;
  userMobile: string;
  userEmail: string;
  videoPlay: boolean;
  video: any;
  counter: number;
  videoLoader: boolean;
  videoPlayerKey?: string | number;
  bubbleHide: boolean;
  companyId: any;
  videoId: any;
  locationPermissionGranted: boolean;
  bubble_time: any;
  dynamic_quiz: any;
  cacheVideoURL: string | null;
  dynamicVideoURL: string;
  last_dynamic_video: any;
  staticVideoCurrentTime: number | string;
  static_adv: boolean;
  geofence: string;
  cacheVideoID: string | number;
  resetSeek: boolean;
  correct_Answer: any;
  staticVideo: any;
  staticPause: boolean;
  allStaticVideos: any;
  allDynamicVideos: any;
  allGeofenes: any;
  cameraOn: any;
  allVideos1: any;
  // Customizable Area End
}
interface SS {
  id: any;
}

export default class DashboardController extends BlockComponent<Props, S, SS> {
  // Customizable Area Start
  apiDashboardItemCallId: string = "";
  quizSaveAnswerId: any = "";
  dynamicStatsId: any = "";
  dashboardApiCallId: string = "";
  apiGetQueryStrinurl: string = "";
  getQuizeDataApiCallId: any = "";
  getQuizesDataApiCallId: any = "";
  unsubscribe: any = null;
  userDataApiCallId: any = "";
  userVideoDataId: any = "";
  lastNumber: number = -1;
  offlineVideoDataId: any = "";
  lat: any = "";
  lng: any = "";
  getBubbleTimerId: any = "";
  getAllVideos1Id: any = "";
  videoCurrentTime: any;
  bubbleUpTime: any;
  isFirstTimeOnline: any;
  stopped: any;
  facesDetected: number;
  totalFacesDetected: any;
  noFacesDetected: any;
  facialCountDataId: any = "";
  tabDataStatusId: any = "";
  facialCountFreq: any;
  offlineVideoNumber: number;
  offlineDynamicNumber: number;
  newFormatQuizDataId: any = "";
  video_Id: any = "";
  company_Id: any = "";
  intervalID: any;
  timeupate: any;
  timeOut: any;
  questionTime: number;
  transLat: any;
  transLong: any;
  cameraTimer: any;
  // Customizable Area End
  
  constructor(props: Props) {
    super(props);
    this.receive = this.receive.bind(this);
    console.disableYellowBox = true;
    // Customizable Area Start
    this.transLat = createRef();
    this.transLong = createRef();
    this.lat = createRef();
    this.lng = createRef();
    this.videoCurrentTime = createRef();
    this.isFirstTimeOnline = createRef();
    this.stopped = createRef();
    this.totalFacesDetected = createRef();
    this.facesDetected = 0;
    this.noFacesDetected = 0;
    this.facialCountFreq = {};
    this.offlineVideoNumber = 0;
    this.video_Id = createRef();
    this.company_Id = createRef();
    this.timeOut = createRef();
    this.intervalID = 0;
    this.timeupate = null;
    this.offlineDynamicNumber = 0;
    this.questionTime = 5;
    this.subScribedMessages = [
      getName(MessageEnum.AccoutLoginSuccess),
      getName(MessageEnum.RestAPIResponceMessage),
      getName(MessageEnum.SessionSaveMessage),
      getName(MessageEnum.SessionResponseMessage),
      getName(MessageEnum.LocationUpdate),
      getName(MessageEnum.GeoFencingEvent)
    ];

    this.state = {
      dashboardData: [],
      errorMsg: "",
      token: "",
      offlineVideo: null,
      quizid: "",
      userName: "",
      userEmail: "",
      userMobile: "",
      videoPlay: false,
      videoLoader: false,
      videoPlayerKey: 2342,
      bubbleHide: false,
      loading: false,
      count: 0,
      quizeData: "",
      sound: false,
      quizQuestionLoaded: false,
      quizQuestionData: null,
      events: [],
      offline: false,
      video: "",
      counter: 0,
      companyId: "",
      videoId: 0,
      locationPermissionGranted: false,
      bubble_time: 30000,
      dynamic_quiz: null,
      cacheVideoURL: null,
      dynamicVideoURL: "",
      last_dynamic_video: false,
      staticVideoCurrentTime: 0,
      static_adv: false,
      cacheVideoID: 1,
      resetSeek: true,
      geofence: "not touhed",
      correct_Answer: [],
      staticVideo: "",
      staticPause: false,
      allStaticVideos: [],
      allDynamicVideos: [],
      allGeofenes: [],
      cameraOn: false,
      allVideos1: []
    };
    // Customizable Area End
    runEngine.attachBuildingBlock(this as IBlock, this.subScribedMessages);
  }

  async componentDidMount() {
    super.componentDidMount();
    this.getToken();
    if (this.isPlatformWeb() === false) {
      this.props.navigation.addListener('willFocus', () => {
        this.getToken();
      });
    }
    // Customizable Area Start
    // Customizable Area End
  }
  
  getToken=()=>{
    const msg: Message = new Message(getName(MessageEnum.SessionRequestMessage));
    this.send(msg);
  }

  getDashboardData(): boolean {
    // Customizable Area Start
    const header = {
      "Content-Type": configJSON.dashboarContentType,
      token: this.state.token
    };
    const requestMessage = new Message(
      getName(MessageEnum.RestAPIRequestMessage)
    );
    this.apiDashboardItemCallId = requestMessage.messageId;
    requestMessage.addData(
      getName(MessageEnum.RestAPIResponceEndPointMessage),
      configJSON.dashboardGetUrl
    );

    requestMessage.addData(
      getName(MessageEnum.RestAPIRequestHeaderMessage),
      JSON.stringify(header)
    );

    requestMessage.addData(
      getName(MessageEnum.RestAPIRequestMethodMessage),
      configJSON.dashboarApiMethodType
    );
    runEngine.sendMessage(requestMessage.id, requestMessage);
    // Customizable Area End
    return true;
  }

  async receive(from: string, message: Message) {
    // Customizable Area Start
    if (
      this.quizSaveAnswerId ===
      message.getData(getName(MessageEnum.RestAPIResponceDataMessage))
    ) {
      const apiResponse = message.getData(
        getName(MessageEnum.RestAPIResponceSuccessMessage)
      );
      console.log("api response from the api", apiResponse);
    }

    if (
      this.newFormatQuizDataId ===
      message.getData(getName(MessageEnum.RestAPIResponceDataMessage))
    ) {
      const apiResponse = message.getData(
        getName(MessageEnum.RestAPIResponceSuccessMessage)
      );
      this.setState({ dynamic_quiz: apiResponse.data.quiz_id });
      let NavigateScreen = apiResponse.data.quiz_type;
      if (NavigateScreen) {
        this.setState({ quizid: apiResponse.data.quiz_id });
        this.setState({
          quizQuestionLoaded: false,
          quizQuestionData: null
        });
        this.props.navigation.navigate(this.GetQuizTypeName(NavigateScreen), {
          quizId: apiResponse.data.quiz_id,
          theme: apiResponse.data.theme,
          question_Time: apiResponse.data.question_time
        });
      }
    }
    if (
      this.facialCountDataId ===
      message.getData(getName(MessageEnum.RestAPIResponceDataMessage))
    ) {
      const apiResponse = message.getData(
        getName(MessageEnum.RestAPIResponceSuccessMessage)
      );
      console.log(apiResponse);
      this.facesDetected = 0;
    }

    if (
      this.getBubbleTimerId ===
      message.getData(getName(MessageEnum.RestAPIResponceDataMessage))
    ) {
      const apiResponse = message.getData(
        getName(MessageEnum.RestAPIResponceSuccessMessage)
      );
      this.setState({ bubble_time: apiResponse.bubble_time });
    }
    if (
      this.getQuizeDataApiCallId ===
      message.getData(getName(MessageEnum.RestAPIResponceDataMessage))
    ) {
      const apiResponse = message.getData(
        getName(MessageEnum.RestAPIResponceSuccessMessage)
      );

      this.setState({ quizeData: apiResponse });
      this.setState({
        quizQuestionLoaded: true,
        quizQuestionData: apiResponse.data
      });
    }
    if (
      this.getQuizesDataApiCallId ===
      message.getData(getName(MessageEnum.RestAPIResponceDataMessage))
    ) {
      const apiResponse = message.getData(
        getName(MessageEnum.RestAPIResponceSuccessMessage)
      );

      if (Array.isArray(apiResponse.data)) {
        let RandomIndex = Math.floor(Math.random() * apiResponse.data.length);

        let NavigateScreen = apiResponse.data[RandomIndex].attributes.quiz_type;
        if (NavigateScreen) {
          this.setState({ quizid: apiResponse.data[RandomIndex].id });
          this.setState({
            quizQuestionLoaded: false,
            quizQuestionData: null
          });
          this.props.navigation.navigate(this.GetQuizTypeName(NavigateScreen), {
            quizId: apiResponse.data[RandomIndex].id,
            theme: apiResponse.data[RandomIndex].attributes.theme
          });
        }
      }
    }
    if (getName(MessageEnum.RestAPIResponceMessage) === message.id) {
      let responseJson = message.getData(
        getName(MessageEnum.RestAPIResponceSuccessMessage)
      );
      if (responseJson && !responseJson.errors && responseJson.data) {
        if (responseJson.data.length === 0) {
          this.setState({
            errorMsg: "Data Not Found",
            loading: false
          });
        } else {
          this.setState({
            dashboardData: responseJson.data,
            errorMsg: "",
            loading: false
          });
        }
      } else {
        let errorReponse = message.getData(
          getName(MessageEnum.RestAPIResponceErrorMessage)
        );
        if (errorReponse === undefined) {
          this.setState({
            errorMsg: "Something went serioulsy wrong",
            loading: false
          });
        } else {
          this.setState({
            errorMsg: errorReponse,
            loading: false
          });
        }
      }
    }
    if (
      this.userDataApiCallId ===
      message.getData(getName(MessageEnum.RestAPIResponceDataMessage))
    ) {
      const apiResponse = message.getData(
        getName(MessageEnum.RestAPIResponceSuccessMessage)
      );
      console.log(apiResponse);
    }
    // Customizable Area End
  }

  // Customizable Area Start
  sendHelloMessage() {
    // console.log("hello  send Hello Message")
    const msg: Message = new Message(getName(MessageEnum.StartGeoFencing));
    msg.properties.text = "hello world";
    this.send(msg);
  }
  GetQuizTypeName = (type: string) => {
    switch (type) {
      case "Quiz-1":
        return "Quiz1";
      case "Quiz-2":
        return "Quiz1";
      case "Quiz-3":
        return "Quiz1";
      case "Quiz-4":
        return "Quiz1";
      case "Quiz-5":
        return "Quiz2";
      case "Quiz-6":
        return "Quiz2";
      case "Quiz-7":
        return "Quiz6";
      case "Quiz-8":
        return "Quiz6";
      case "Quiz-12":
        return "Quiz6";
      case "Quiz-9":
        return "Quiz5";
      case "Quiz-10":
        return "Quiz5";
      case "Quiz-11":
        return "Quiz5";
      case "Quiz-14":
        return "Quiz4";
      case "Quiz-13":
        return "Quiz3";

      default:
        return null;
    }
  };
  fetchQuizeData = (id: any) => {
    const headers = {
      "Content-Type": configJSON.dashboarContentType,
      token: configJSON.staticToken
    };
    const getQuizeDataValidation = new Message(
      getName(MessageEnum.RestAPIRequestMessage)
    );
    this.getQuizeDataApiCallId = getQuizeDataValidation.messageId;

    getQuizeDataValidation.addData(
      getName(MessageEnum.RestAPIResponceEndPointMessage),
      "bx_block_cfquiztrivia/quizzes/" + id
    );
    getQuizeDataValidation.addData(
      getName(MessageEnum.RestAPIRequestHeaderMessage),
      JSON.stringify(headers)
    );

    getQuizeDataValidation.addData(
      getName(MessageEnum.RestAPIRequestMethodMessage),
      "GET"
    );

    runEngine.sendMessage(getQuizeDataValidation.id, getQuizeDataValidation);
  };
  fetchallQuizData = () => {
    const headers = {
      "Content-Type": configJSON.dashboarContentType,
      token: configJSON.staticToken
    };
    const getQuizesDataValidation = new Message(
      getName(MessageEnum.RestAPIRequestMessage)
    );
    this.getQuizesDataApiCallId = getQuizesDataValidation.messageId;

    getQuizesDataValidation.addData(
      getName(MessageEnum.RestAPIResponceEndPointMessage),
      "bx_block_cfquiztrivia/quizzes"
    );

    getQuizesDataValidation.addData(
      getName(MessageEnum.RestAPIRequestHeaderMessage),
      JSON.stringify(headers)
    );

    getQuizesDataValidation.addData(
      getName(MessageEnum.RestAPIRequestMethodMessage),
      "GET"
    );

    runEngine.sendMessage(getQuizesDataValidation.id, getQuizesDataValidation);
  };
  sendUserData = async (id: any) => {
    let { advertisingId } = await RNAdvertisingId.getAdvertisingId();
    const UserDataValidation = new Message(
      getName(MessageEnum.RestAPIRequestMessage)
    );
    this.userDataApiCallId = UserDataValidation.messageId;

    UserDataValidation.addData(
      getName(MessageEnum.RestAPIResponceEndPointMessage),
      "bx_block_cfquiztrivia/participents"
    );

    const header = {
      "Content-Type": "multipart/form-data",
      token: configJSON.staticToken
    };
    UserDataValidation.addData(
      getName(MessageEnum.RestAPIRequestHeaderMessage),
      JSON.stringify(header)
    );
    UserDataValidation.addData(
      getName(MessageEnum.RestAPIRequestMethodMessage),
      "POST"
    );

    let formdata = new FormData();
    formdata.append("name", this.state.userName);
    formdata.append("email", this.state.userEmail);
    formdata.append("full_phone_number", this.state.userMobile);
    formdata.append("quiz_id", id);
    formdata.append("unique_serial_no", advertisingId);

    UserDataValidation.addData(
      getName(MessageEnum.RestAPIRequestBodyMessage),
      formdata
    );
    runEngine.sendMessage(UserDataValidation.id, UserDataValidation);
    console.log("quiz idf----from api hit", id);

    if (
      this.state.userEmail.length > 0 ||
      this.state.userMobile.length > 0 ||
      this.state.userName.length > 0
    ) {
      Toast.show("Your details are successfully submitted.", Toast.SHORT);
      this.setState({ userName: "", userMobile: "", userEmail: "" });
      this.props.navigation.navigate("Home");
    } else {
      this.setState({ userName: "", userMobile: "", userEmail: "" });
      this.props.navigation.navigate("Home");
    }
  };

  updateTimer = () => {
    const header = {
      "Content-Type": configJSON.dashboarContentType,
      token: configJSON.staticToken
    };
    const getBubbleDataValidation = new Message(
      getName(MessageEnum.RestAPIRequestMessage)
    );
    this.getBubbleTimerId = getBubbleDataValidation.messageId;

    getBubbleDataValidation.addData(
      getName(MessageEnum.RestAPIResponceEndPointMessage),
      "bx_block_cfquiztrivia/get_bubble_time"
    );
    getBubbleDataValidation.addData(
      getName(MessageEnum.RestAPIRequestHeaderMessage),
      JSON.stringify(header)
    );

    getBubbleDataValidation.addData(
      getName(MessageEnum.RestAPIRequestMethodMessage),
      "GET"
    );

    runEngine.sendMessage(getBubbleDataValidation.id, getBubbleDataValidation);
  };
  locationBasedQuiz = () => {
    const headers = {
      "Content-Type": configJSON.dashboarContentType,
      token: configJSON.staticToken
    };

    const Quizdata = new Message(getName(MessageEnum.RestAPIRequestMessage));
    this.newFormatQuizDataId = Quizdata.messageId;

    Quizdata.addData(
      getName(MessageEnum.RestAPIRequestHeaderMessage),
      JSON.stringify(headers)
    );
    if (this.state.dynamic_quiz == null) {
      Quizdata.addData(
        getName(MessageEnum.RestAPIResponceEndPointMessage),
        `bx_block_cfquiztrivia/sequence_wise_quiz?quiz_id=`
      );
    } else {
      Quizdata.addData(
        getName(MessageEnum.RestAPIResponceEndPointMessage),
        `bx_block_cfquiztrivia/sequence_wise_quiz?quiz_id=${
          this.state.dynamic_quiz
        }`
      );
    }

    Quizdata.addData(getName(MessageEnum.RestAPIRequestMethodMessage), "GET");

    runEngine.sendMessage(Quizdata.id, Quizdata);
  };
  saveUserQuizAnswers = async (id: any, answers: any) => {
    let { advertisingId } = await RNAdvertisingId.getAdvertisingId();
    const headers = {
      "Content-Type": configJSON.dashboarContentType,
      token: configJSON.staticToken
    };
    let body: any = {};
    body.quiz_id = id;
    body.user_answers = answers;
    body.unique_serial_no = advertisingId;

    const QuizAnswerData = new Message(
      getName(MessageEnum.RestAPIRequestMessage)
    );

    this.quizSaveAnswerId = QuizAnswerData.messageId;

    QuizAnswerData.addData(
      getName(MessageEnum.RestAPIResponceEndPointMessage),
      "bx_block_cfquiztrivia/quizzes/save_answers"
    );

    QuizAnswerData.addData(
      getName(MessageEnum.RestAPIRequestBodyMessage),
      JSON.stringify(body)
    );
    QuizAnswerData.addData(
      getName(MessageEnum.RestAPIRequestHeaderMessage),
      JSON.stringify(headers)
    );
    QuizAnswerData.addData(
      getName(MessageEnum.RestAPIRequestMethodMessage),
      "POST"
    );
    runEngine.sendMessage(QuizAnswerData.id, QuizAnswerData);
  };
  // Customizable Area End

}
