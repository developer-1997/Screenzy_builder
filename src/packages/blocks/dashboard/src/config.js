Object.defineProperty(exports, "__esModule", {
  value: true
});

// Customizable Area Start
exports.dashboardGetUrl = "/dashboard/dashboards";
exports.dashboarContentType = "application/json";
exports.dashboarApiMethodType = "GET";
exports.dashboardHost = "<calculated when request is sent>";
exports.dashboarUserAgent = "PostmanRuntime/7.26.5";
exports.dashboarAccept = "*/*";
exports.dashboarAcceptEncoding = "gzip, deflate, br";
exports.dashboarConnection = "keep-alive";
exports.dashboartoken = "";
exports.labelTitleText = "dashboard";
exports.labelBodyText = "dashboard Body";
exports.liveVideoURL =  "https://commondatastorage.googleapis.com/gtv-videos-bucket/sample/ForBiggerFun.mp4";
exports.offlineVideoURL = "https://samplelib.com/lib/preview/mp4/sample-10s.mp4";
exports.dummyofflineVideoURL = "https://samplelib.com/lib/preview/mp4/sample-5s.mp4";
exports.staticToken = "tyugiojpkiy8ucfghvjbuy76tfygui86t7yuhu7vbnubhhjihbjihjghjkiou897ytfyghjkiou8976tfygvhjbkiu8y76t5tfychjbuhy"
//Customizable Area End