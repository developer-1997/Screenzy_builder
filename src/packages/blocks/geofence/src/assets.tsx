export const AddIcon = require("../assets/add.svg") 