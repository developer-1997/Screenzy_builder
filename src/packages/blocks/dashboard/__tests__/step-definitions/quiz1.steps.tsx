import { defineFeature, loadFeature } from "jest-cucumber";
import { shallow, ShallowWrapper } from "enzyme";

import * as helpers from "../../../../framework/src/Helpers";
import React from "react";
import Quiz1 from "../../src/quizzes/Quiz1";
import { runEngine } from "../../../../framework/src/RunEngine";
import { Message } from "../../../../framework/src/Message";
import MessageEnum, {
  getName
} from "../../../../framework/src/Messages/MessageEnum";
import { View } from "react-native";

const mockNavigation = {
  state: {
    params: {
      Quizid: "135"
    }
  },
  replace: jest.fn()
};

const screenProps = {
  navigation: mockNavigation,
  id: "Quiz1"
};
const feature = loadFeature("./__tests__/features/quiz1-scenario.feature");

const quizQuestionDataMock = {
  id: "135",
  type: "quiz",
  attributes: {
    theme: "black",
    quiz_type: "Quiz-1",
    questions: [
      {
        id: "169",
        type: "question",
        attributes: {
          question_type: "title",
          title:
            "Do you like to get 20% discount on your Hoichoi subscription? ",
          question_image:
            "https://screenzycommercialsllp-125263-ruby.b125263.stage.eastus.az.svc.builder.ai/rails/active_storage/blobs/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBdTBEIiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--ee27116e36d3d2cd71f0266a6a233e8cdf723451/Hoichoi-Logos-3.png",
          options: [
            {
              id: "591",
              type: "option",
              attributes: {
                option_type: "title",
                option_title: "Yes, 100%",
                is_correct: true,
                option_image: null
              }
            },
            {
              id: "592",
              type: "option",
              attributes: {
                option_type: "title",
                option_title: "No",
                is_correct: false,
                option_image: null
              }
            },
            {
              id: "593",
              type: "option",
              attributes: {
                option_type: "title",
                option_title: "I already have Subscription ",
                is_correct: true,
                option_image: null
              }
            }
          ]
        }
      }
    ]
  }
};
defineFeature(feature, test => {
  beforeEach(() => {
    jest.resetModules();
    jest.doMock("react-native", () => ({ Platform: { OS: "web" } }));
    jest.spyOn(helpers, "getOS").mockImplementation(() => "web");
  });

  test("User navigates to quiz1", ({ given, when, then }) => {
    let quiz1Wrapper: ShallowWrapper;
    let instance: Quiz1;
    given("I am a User loading quiz1", () => {
      quiz1Wrapper = shallow(<Quiz1 {...screenProps} />);
      expect(quiz1Wrapper).toBeTruthy();
      expect(quiz1Wrapper).toMatchSnapshot();
    });

    when("I navigate to the quiz1", () => {
      instance = quiz1Wrapper.instance() as Quiz1;
      expect(quiz1Wrapper).toBeTruthy();
      expect(quiz1Wrapper).toMatchSnapshot();
    });
    then("quiz1 will load with out errors", () => {
      expect(quiz1Wrapper).toBeTruthy();
      expect(quiz1Wrapper).toMatchSnapshot();
    });

    then("Quiz1 will display messages", () => {
      const tokenMsg: Message = new Message(
        getName(MessageEnum.SessionResponseMessage)
      );
      tokenMsg.addData(getName(MessageEnum.SessionResponseToken), "TOKEN");
      runEngine.sendMessage("Unit Test", tokenMsg);

      const apiMsg: Message = new Message(
        getName(MessageEnum.RestAPIResponceMessage)
      );
      apiMsg.addData(getName(MessageEnum.RestAPIResponceSuccessMessage), {
        data: [
          {
            id: 1,
            type: "quiz1",
            attributes: {
              id: 1,
              title: "Quiz1_title_1",
              value: 1,
              created_at: "2023-04-02T17:10:08.139Z",
              updated_at: "2023-04-02T17:10:08.139Z"
            }
          },
          {
            id: 2,
            type: "quiz1",
            attributes: {
              id: 2,
              title: "Quiz1 5",
              value: 5,
              created_at: "2023-04-02T17:10:36.867Z",
              updated_at: "2023-04-02T17:10:36.867Z"
            }
          }
        ]
      });
      runEngine.sendMessage("Unit Test", apiMsg);

      expect(quiz1Wrapper).toBeTruthy();
      expect(quiz1Wrapper).toMatchSnapshot();
    });
    then("Quiz1 will display notifcation if no messages", () => {
      const apiNoItemsMsg: Message = new Message(
        getName(MessageEnum.RestAPIResponceMessage)
      );
      apiNoItemsMsg.addData(
        getName(MessageEnum.RestAPIResponceSuccessMessage),
        { data: [] }
      );
      runEngine.sendMessage("Unit Test", apiNoItemsMsg);

      expect(quiz1Wrapper).toBeTruthy();
      expect(quiz1Wrapper).toMatchSnapshot();
    });
    then("Quiz1 will display notifcation if API failure", () => {
      const apiNoItemsMsg: Message = new Message(
        getName(MessageEnum.RestAPIResponceMessage)
      );
      apiNoItemsMsg.addData(
        getName(MessageEnum.RestAPIResponceSuccessMessage),
        { data: [] }
      );
      runEngine.sendMessage("Unit Test", apiNoItemsMsg);

      const apiErrorResponceMsg: Message = new Message(
        getName(MessageEnum.RestAPIResponceMessage)
      );
      apiErrorResponceMsg.addData(
        getName(MessageEnum.RestAPIResponceSuccessMessage),
        { errors: "Error" }
      );
      runEngine.sendMessage("Unit Test", apiErrorResponceMsg);

      const apiFailedErrorResponceMsg: Message = new Message(
        getName(MessageEnum.RestAPIResponceMessage)
      );
      runEngine.sendMessage("Unit Test", apiFailedErrorResponceMsg);

      expect(quiz1Wrapper).toBeTruthy();
      expect(quiz1Wrapper).toMatchSnapshot();
    });

    then("I can leave the screen with out errors", () => {
      instance.componentWillUnmount();
      expect(quiz1Wrapper).toBeTruthy();
      expect(quiz1Wrapper).toMatchSnapshot();
    });
  });

  test("test for show quiz1 modal", ({ given, when, then }) => {
    let quizWrapper: ShallowWrapper;
    let instance: Quiz1;

    given("I am a User attempting to quiz1", () => {
      quizWrapper = shallow(<Quiz1 {...screenProps} />);
      expect(quizWrapper).toBeTruthy();
    });

    when("I am user try to attempting quiz1", () => {
      instance = quizWrapper.instance() as Quiz1;
      instance.setState({ quizQuestionLoaded: true });
    });

    then("I can attempt the quiz1", () => {
      expect(quizWrapper.contains(<View testID="showQuiz" />)).toBe(false);
    });
  });

  test("test for you win quiz1", ({ given, when, then }) => {
    let quizWrapper1: ShallowWrapper;
    let instance: Quiz1;

    given("I am a User navigate to quiz1 win", () => {
      quizWrapper1 = shallow(<Quiz1 {...screenProps} />);
      expect(quizWrapper1).toBeTruthy();
    });

    when("I am user try to load quiz1 win", () => {
      instance = quizWrapper1.instance() as Quiz1;
      instance.setState({
        quizQuestionData: quizQuestionDataMock,
        index: 0
      });
    });

    then("I win the quiz1", () => {
      instance.handelOnComplete();
    });
  });

  test("test for you loss quiz1", ({ given, when, then }) => {
    let quizWrapper1: ShallowWrapper;
    let instance: Quiz1;

    given("I am a User navigate to quiz1 loss", () => {
      quizWrapper1 = shallow(<Quiz1 {...screenProps} />);
      expect(quizWrapper1).toBeTruthy();
    });

    when("I am user try to load quiz1 loss", () => {
      instance = quizWrapper1.instance() as Quiz1;
      instance.setState({
        index: 0
      });
    });

    then("I loss the quiz1", () => {
      instance.handelOnComplete();
    });
  });

  test("test for clicked quiz1 right option", ({ given, when, then }) => {
    let quizWrapper1: ShallowWrapper;
    let instance: Quiz1;

    given("I can see quiz1 options", () => {
      quizWrapper1 = shallow(<Quiz1 {...screenProps} />);
      expect(quizWrapper1).toBeTruthy();
    });

    when("I am user try to click one option of quiz1", () => {
      instance = quizWrapper1.instance() as Quiz1;
      instance.setState({
        index: 0,
        timerClock: 15,
        quizQuestionData: quizQuestionDataMock
      });
      instance.handelOnComplete();
    });
    then("I can clicked right answer", () => {
      instance.handelOptionPress(
        quizQuestionDataMock.attributes.questions[0].attributes.options[0],
        0
      );
    });
  });

  test("test for clicked quiz1 wrong options", ({ given, when, then }) => {
    let quizWrapper1: ShallowWrapper;
    let instance: Quiz1;

    given("I can see quiz1 options", () => {
      quizWrapper1 = shallow(<Quiz1 {...screenProps} />);
      expect(quizWrapper1).toBeTruthy();
    });

    when("I am user try to click one option of quiz1", () => {
      instance = quizWrapper1.instance() as Quiz1;
      instance.setState({
        index: 0,
        timerClock: 15,
        quizQuestionData: quizQuestionDataMock
      });
      instance.handelOnComplete();
    });
    then("I can clicked wrong answer", () => {
      instance.handelOptionPress(
        quizQuestionDataMock.attributes.questions[0].attributes.options[1],
        0
      );
    });
  });

  test("test for clicked quiz1 right option 2", ({ given, when, then }) => {
    let quizWrapper1: ShallowWrapper;
    let instance: Quiz1;

    given("I can see quiz1 options", () => {
      quizWrapper1 = shallow(<Quiz1 {...screenProps} />);
      expect(quizWrapper1).toBeTruthy();
    });

    when("I am user try to click one option of quiz1", () => {
      instance = quizWrapper1.instance() as Quiz1;
      instance.setState({
        index: 0,
        timerClock: 15,
        quizQuestionData: quizQuestionDataMock
      });
      instance.handelOnComplete();
    });
    then("I  clicked right  answer 2", () => {
      instance.handelOptionPress(
        quizQuestionDataMock.attributes.questions[0].attributes.options[2],
        0
      );
    });
  });
});
