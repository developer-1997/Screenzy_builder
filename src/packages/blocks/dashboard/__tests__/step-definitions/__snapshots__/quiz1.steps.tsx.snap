// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`quiz1 User navigates to quiz1 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#000",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../packages/blocks/dashboard/assets/mainlogo.png",
      }
    }
    style={
      Object {
        "height": 196,
        "resizeMode": "contain",
        "width": 196,
      }
    }
  />
  <ActivityIndicator
    color="#fff"
    size="large"
  />
</View>
`;

exports[`quiz1 User navigates to quiz1 2`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#000",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../packages/blocks/dashboard/assets/mainlogo.png",
      }
    }
    style={
      Object {
        "height": 196,
        "resizeMode": "contain",
        "width": 196,
      }
    }
  />
  <ActivityIndicator
    color="#fff"
    size="large"
  />
</View>
`;

exports[`quiz1 User navigates to quiz1 3`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#000",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../packages/blocks/dashboard/assets/mainlogo.png",
      }
    }
    style={
      Object {
        "height": 196,
        "resizeMode": "contain",
        "width": 196,
      }
    }
  />
  <ActivityIndicator
    color="#fff"
    size="large"
  />
</View>
`;

exports[`quiz1 User navigates to quiz1 4`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#000",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../packages/blocks/dashboard/assets/mainlogo.png",
      }
    }
    style={
      Object {
        "height": 196,
        "resizeMode": "contain",
        "width": 196,
      }
    }
  />
  <ActivityIndicator
    color="#fff"
    size="large"
  />
</View>
`;

exports[`quiz1 User navigates to quiz1 5`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#000",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../packages/blocks/dashboard/assets/mainlogo.png",
      }
    }
    style={
      Object {
        "height": 196,
        "resizeMode": "contain",
        "width": 196,
      }
    }
  />
  <ActivityIndicator
    color="#fff"
    size="large"
  />
</View>
`;

exports[`quiz1 User navigates to quiz1 6`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#000",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../packages/blocks/dashboard/assets/mainlogo.png",
      }
    }
    style={
      Object {
        "height": 196,
        "resizeMode": "contain",
        "width": 196,
      }
    }
  />
  <ActivityIndicator
    color="#fff"
    size="large"
  />
</View>
`;

exports[`quiz1 User navigates to quiz1 7`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#000",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../packages/blocks/dashboard/assets/mainlogo.png",
      }
    }
    style={
      Object {
        "height": 196,
        "resizeMode": "contain",
        "width": 196,
      }
    }
  />
  <ActivityIndicator
    color="#fff"
    size="large"
  />
</View>
`;
